nondet str issue 3284

sergillam opened the pull request:
Fixes #3284

Commit:
[python-frontend] - add supports expressions in pointer offset

Commit:
[python-frontend] - update in test

lucasccordeiro committed:
Update check_python_tests.sh

lucasccordeiro committed:
Rename test descriptor from CORE to KNOWNBUG

LukeW1999 commented:
Hi @sergillam 
Please have a look at this example:
```
foo = nondet_str()

if foo == "":
    foo = "foo"

assert foo == "foo"
```

```
CORE
main.py
--unwind 4
^VERIFICATION FAILED$
```

Commit:
[python-frontend]  -fix nondet_str() symbolic semantics in conditionals

Skip static resolution for string comparisons to preserve symbolic path
semantics, preventing nondet values from being replaced by cached literals.

sergillam committed:
Apply automatic changes

lucasccordeiro committed:
Merge branch 'master' into feat/python-nondet_str-issue-3284
